plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/update/VersionUpdateDialog.java

```diff
@@ -253,104 +251,107 @@ protected void buttonPressed(int buttonId)
         if (dontShowAgainCheck != null && dontShowAgainCheck.getSelection()) {
             CoreApplicationActivator.getDefault().getPreferenceStore().setValue("suppressUpdateCheck." + newVersion.getPlainVersion(), true);
         }
-        if (buttonId == INFO_ID) {
-            ShellUtils.launchProgram(newVersion.getBaseURL());
-        } else if (buttonId == UPGRADE_ID) {
-            final PlatformInstaller installer = getPlatformInstaller();
-            if (installer != null) {
-                final AbstractJob job = new AbstractJob("Downloading installation file") {
+        switch (buttonId) {
+            case INFO_ID -> showInfo();
+            case CHECK_EA_ID -> showEarlyAccess();
+            case UPGRADE_ID -> runUpgrade();
+        }
+        close();
+    }
+
+    private void showInfo() {
+        ShellUtils.launchProgram(newVersion.getBaseURL());
+    }
+
+    private void showEarlyAccess() {
+        if (CommonUtils.isNotEmpty(earlyAccessURL)) {
+            ShellUtils.launchProgram(earlyAccessURL);
+        }
+    }
+
+    private void runUpgrade() {
+        final PlatformInstaller installer = getPlatformInstaller();
+        if (installer == null) {
+            ShellUtils.launchProgram(getDownloadPageURL(newVersion));
+            return;
+        }
+        final AbstractJob job = new AbstractJob("Downloading installation file") {
+            @Override
+            protected IStatus run(DBRProgressMonitor monitor) {
+                final ApplicationDescriptor app = ApplicationRegistry.getInstance().getApplication();
+                final Path folder;
+                final Path file;
+
+                try {
+                    final String executable = installer.getExecutableName(app);
+
+                    folder = Files.createTempDirectory(executable);
+                    file = Files.createFile(folder.resolve(executable));
+
+                    log.debug("Downloading installation file to " + file);
+                    WebUtils.downloadRemoteFile(monitor, "Obtaining installer", getDownloadURL(app, installer, newVersion), file, null);
+                } catch (IOException e) {
+                    return GeneralUtils.makeErrorStatus(CoreMessages.dialog_version_update_downloader_error_cannot_download, e);
+                } catch (InterruptedException e) {
+                    log.debug("Canceled by user", e);
+                    return Status.OK_STATUS;
+                }
+
+                final boolean confirmed = UIUtils.confirmAction(
+                        CoreMessages.dialog_version_update_downloader_title,
+                        NLS.bind(CoreMessages.dialog_version_update_downloader_confirm_install, app.getName())
```

Comment:
We use 4-spaces continuation indent

```suggestion
                    CoreMessages.dialog_version_update_downloader_title,
                    NLS.bind(CoreMessages.dialog_version_update_downloader_confirm_install, app.getName())
```

Reply:
Oh my... I missed it. Thank you for careful review!